app/javascript/packages/document-capture/components/form-steps.jsx

```diff
@@ -187,6 +212,9 @@ function FormSteps({ steps = [], onComplete = () => {}, initialValues = {}, init
         onChange={(nextValuesPatch) => {
           setValues((prevValues) => ({ ...prevValues, ...nextValuesPatch }));
         }}
+        registerField={(fieldName) => (fieldNode) => {
+          fields.current[fieldName] = fieldNode;
+        }}
```

Comment:
this is a little tricky for me to follow, WDYT of adding braces to the outer function?

```suggestion
        registerField={(fieldName) => {
          return (fieldNode) => { fields.current[fieldName] = fieldNode; }
        }}
```

Reply:
Funny enough, in trying to implement this, it's actually _enforced_ to be this way:

```
$ yarn lint app/javascript/packages/document-capture/components/form-steps.jsx
yarn run v1.22.4
$ eslint app spec babel.config.js --ext .js,.jsx app/javascript/packages/document-capture/components/form-steps.jsx

/Users/andrewmduthie/Documents/Code/identity-idp/app/javascript/packages/document-capture/components/form-steps.jsx
  215:39  error  Unexpected block statement surrounding arrow body; move the returned value immediately after the `=>`  arrow-body-style

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

Via:

- https://github.com/airbnb/javascript#arrows--implicit-return
- https://eslint.org/docs/rules/arrow-body-style.html

We could choose to disable this rule. Personally, I don't think it reads especially clear in any of the possible implementations 😅  But I've become accustomed enough to seeing the chained arrow expressions.

Reply:
After closer inspection, this might need a bit of work, since the [callback ref](https://reactjs.org/docs/refs-and-the-dom.html#callback-refs) we're passing will be a unique function reference each time (`ref !== nextRef`), which from what I can tell will cause React to recycle on each render (call `callback(null)`, `callback(nextElement)`). This is why the [`setVideoRef` callback ref in `SelfieCapture`](https://github.com/18F/identity-idp/blob/3e7174e0ceb06f4f2f9cc5eebfea40fed8a883d4/app/javascript/packages/document-capture/components/selfie-capture.jsx#L31) was created as a memoized function, in order to keep the reference constant render to render.

Reply:
ad191b2 includes revisions for both:

1. Ensuring ref callback is consistent between renders.
2. Formatting `registerField` in a way that's not a series of chained arrow functions

Reply:
seems not worth updating the linter